### PR TITLE
Podman support

### PR DESCRIPTION
**Pull request by edigaryev**
`TestResourceLimits` had to be disabled when running with Podman backend due to https://github.com/containers/podman/issues/7959.

There are some things that still need to be done before this gets merged:

* Podman socket location auto-detection
* Rename `DockerOptions` to `ContainerOptions` and other de-Dockerizations (e.g. `'docker pull' succeeded in 4.2s`)

### CODE REVIEW

**Comment by edigaryev** [comment]
@fkorotkov what do you think about moving `podmanapi` package to a separate repository to avoid huge diff?

**Comment by edigaryev** [comment]
>Rename `DockerOptions` to `ContainerOptions` and other de-Dockerizations (e.g. `'docker pull' succeeded in 4.2s`)

https://github.com/cirruslabs/cirrus-cli/pull/151/commits/be8883efde7e68397e4e38652f2d963bbebd0eb0 and https://github.com/cirruslabs/cirrus-cli/pull/151/commits/43f152d7fd2c24408c752233706b88969d8f86fb.

>Podman socket location auto-detection

I'm reading the [Podman's API document](https://github.com/containers/podman/blob/master/docs/tutorials/podman-derivative-api.md) and thinking that we could probably combine [Using the REST API](https://github.com/containers/podman/blob/master/docs/tutorials/podman-derivative-api.md#using-the-rest-api) with the [Running as a subprocess](https://github.com/containers/podman/blob/master/docs/tutorials/podman-derivative-api.md#running-as-a-subprocess) to get the best of both worlds.

By spawning a `podman system service` process in `containerbackend.NewPodman()`, we could get a predictable socket location and provide an experience similar to what Docker does when everything works out of the box.

An additional `--podman-binary` flag can be provided for development purposes in case putting a custom-built `podman` binary in `PATH` is not feasible for some reason.

**Review by fkorotkov — APPROVED** [review]
Looks amazing! What do you think about an option to pass backend as an argument to `run` command? Something like `cirrus run --backend=podman`. Or we can do in the follow up PR about running podman binary instead of relying on a running podman service.

File: .cirrus.yml [hunk 1]
```diff
@@ -10,21 +10,42 @@ task:
...
   test_script:
     - wget --no-verbose -O - https://golang.org/dl/go1.15.linux-amd64.tar.gz | tar -C /usr/local -xz
     - export PATH=$PATH:/usr/local/go/bin
     - go test ./...
   env:
     HOME: /root
 
+docker_builder:
+  name: Test (Podman)
+  alias: test-podman
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply .cirrus.yml]: Renamed in https://github.com/cirruslabs/cirrus-cli/pull/151/commits/e9083930ae9c67bed5d535ab6854e7bb0bbc30d4.

**Comment by edigaryev** [comment]
> What do you think about an option to pass backend as an argument to `run` command? Something like `cirrus run --backend=podman`.

Added in https://github.com/cirruslabs/cirrus-cli/pull/151/commits/b3b9add732a53475e65a3d93611cad00f2b4afcd.

>Or we can do in the follow up PR about running podman binary instead of relying on a running podman service.

This was already has been done here: https://github.com/cirruslabs/cirrus-cli/pull/151/commits/a4d143cd7574b630e86d56a9dbb08327cbde6572

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**